File: 프로그래머스/unrated/140107. 점 찍기/점 찍기.py
def solution(k, d):
    answer = 0
    x_index = 0
    d2 = d**2
    for i in range(d//k + 1):
        y_d = int((d2 - (i*k)**2)**0.5)
        answer += y_d // k + 1
    return answer

File: 프로그래머스/2/140107. 점 찍기/점 찍기.java
import java.util.*;

class Solution {
    long answer;
    int k, d;
    public long solution(int k, int d) {
        init(k, d);
        solve();
        return answer;
    }
    
    void init(int k, int d) {
        answer = 0;
        this.k = k;
        this.d = d;
    }
    
    void solve() {
        long powered = (long) Math.pow(d, 2);
        for (int i = 0; i <= d; i += k) {
            long diff = powered - (long) Math.pow(i, 2);
            int sqrt = (int) Math.sqrt(diff) / k;
            answer += sqrt + 1;
        }
    }
}